pkg/stanza/operator/transformer/recombine/recombine.go

```diff
@@ -150,7 +152,14 @@ type Transformer struct {
 	sourceIdentifier    entry.Field
 
 	sync.Mutex
-	batchMap map[string][]*entry.Entry
+	batchMap   map[string]*sourceBatch
+	maxLogSize int64
+}
+
+// sourceBatch contains the status info of a batch
+type sourceBatch struct {
+	entries    []*entry.Entry
```

Comment:
```suggestion
	base    *entry.Entry // first or last, according to overwriteWithOldest
	lastObserved time.Time
```

I think we have an opportunity to reduce the memory footprint now.

Reply:
yeah, but there is one thing to note:
The logic of flushUncombined should be changed. With []*entry.Entry, flushUncombined will flush all entries individualy. With base entry, We can only send the aggregated entry.

Reply:
OK, so if we don't want to make the change, the memory footprint can't be reduced, we need all the entries.

Reply:
Thanks @yutingcaicyt, I appreciate you being flexible on that.

Reply:
Thanks a lot. Are there any other problems? I'm sorry to make some mistakes in my first pr.